language/move-analyzer/src/symbols.rs

```diff
@@ -2190,6 +2198,228 @@ pub fn on_hover_request(context: &Context, request: &Request, symbols: &Symbols)
     );
 }
 
+/**
+ * Returns a tuple where the first value is the index of the text info of the function name and the second value is the text info of the function
+ */
+fn find_closet_fn_in_text_info_vec<'a>(
+    start_idx: usize,
+    text_info_vec: &'a Vec<TextInfo>,
+) -> Option<(usize, TextInfo<'a>)> {
+    let mut parenthesis_delta = 0;
+
+    // Plus 1 to include the character at the starting index
+    for i in (0..start_idx + 1).rev() {
+        let text_info = text_info_vec.get(i).unwrap();
+
+        match text_info.text {
+            "(" => {
+                if parenthesis_delta == 0 {
+                    // The identifier to the left of the left parenthesis is the function.
+                    match text_info_vec.get(i - 1) {
+                        Some(text_info) => {
+                            return Some((i - 1, *text_info));
+                        }
+                        None => {
+                            return None;
+                        }
+                    }
+                } else {
+                    parenthesis_delta -= 1;
+                }
+            }
+            ")" => {
+                parenthesis_delta += 1;
+            }
+            _ => {}
+        }
+    }
+
+    None
+}
+
+/**
+ * Find the position of the active parameter given a vector of parsed tokens
```

Comment:
Please add more explanation of what parsed tokens contains (what kind of line was parsed to obtain the tokens).

Reply:
done